Assets/Scripts/Models/Area/RoomBehavior.cs

```diff
@@ -387,16 +387,15 @@ public RoomBehavior Clone()
         public void Control(Room room)
         {
             this.Room = room;
-            List<Tile> innerTiles = room.GetInnerTiles();
-            List<Tile> borderTiles = room.GetBorderingTiles();
-
-            List<Tile> allTiles = innerTiles.Union(borderTiles).ToList();
+            HashSet<Tile> innerTiles = room.GetInnerTiles();
+            HashSet<Tile> borderTiles = room.GetBoundaryTiles();
+            innerTiles.UnionWith(borderTiles);
 
             foreach (FurnitureRequirement requirement in requiredFurniture)
             {
                 string furnitureKey = requirement.type ?? requirement.typeTag;
                 ControlledFurniture.Add(furnitureKey, new List<Furniture>());
-                foreach (Tile tile in allTiles.FindAll(tile => (tile.Furniture != null && (tile.Furniture.Type == requirement.type || tile.Furniture.HasTypeTag(requirement.typeTag)))))
+                foreach (Tile tile in innerTiles.Where(tile => (tile.Furniture != null && (tile.Furniture.Type == requirement.type || tile.Furniture.HasTypeTag(requirement.typeTag)))))
```

Comment:
FindAll is only faster for lists since it is a list implemented method, where as Where is a linq method which makes it faster for hash set, so while you could say this results in a slower procedure it cuts down on an entire list creation every frame (and union with is much faster than union) so overall it results being about the same speed wise but more memory efficient due to the hashset union.  So basically its a side-effect of hashset.